src/pymor/reductors/basic.py

```diff
@@ -152,30 +154,152 @@ def extend_basis(self, U, method='gram_schmidt', pod_modes=1, pod_orthonormalize
             Raised when the selected extension method does not yield a basis of increased
             dimension.
         """
-        assert method in ('trivial', 'gram_schmidt', 'pod')
-
-        basis_length = len(self.RB)
-
-        if method == 'trivial':
-            remove = set()
-            for i in range(len(U)):
-                if np.any(almost_equal(U[i], self.RB)):
-                    remove.add(i)
-            self.RB.append(U[[i for i in range(len(U)) if i not in remove]],
-                           remove_from_other=(not copy_U))
-        elif method == 'gram_schmidt':
-            self.RB.append(U, remove_from_other=(not copy_U))
-            gram_schmidt(self.RB, offset=basis_length, product=self.product, copy=False)
-        elif method == 'pod':
-            if self.product is None:
-                U_proj_err = U - self.RB.lincomb(U.dot(self.RB))
+        extend_basis(self.RB, U, self.product, method=method, pod_modes=pod_modes,
+                     pod_orthonormalize=pod_orthonormalize, copy_U=copy_U)
+
+
+class GenericPGReductor(BasicInterface):
+    """Generic Petrov-Galerkin reductor.
+
+    Replaces each |Operator| of the given |Discretization| with the projection
+    onto the span of the given projection matrices.
+
+    Parameters
+    ----------
+    d
+        The |Discretization| which is to be reduced.
+    V
+        |VectorArray| containing the right projection matrix.
+    W
+        |VectorArray| containing the left projection matrix.
+    biorthogonal_product
+        Key of the operator in d.operators() used as the product to
+        biorthogonalize V and W.
```

Comment:
This functionality does not quite match what we have for `GenericRBReductor`. I think we should make both classes feel more consistent. But this can also happen after the merge.